### ISSUE TITLE
🐛 TypeScript Class With Decorator Not Exported

### ISSUE THREAD

**Issue by Sascha8a** [rome/tools#1021]
Rome shows `This file doesn't have any exports`, when in fact there are exports present.

## Environment

```
$ rome rage --summary
Environment Variables
  Object {
    COLORTERM: "truecolor"
    LANG: "en_US.UTF-8"
    PATH: "/home/sascha/.sdkman/candidates/java/current/bin:/home/sascha/perl5/bin:/home/sascha/.pyenv/shims:/home/sascha/.pyenv/bin:/home/sascha/.nvm/versions/node/v12.18.3/bin:/home/sascha/.pip-apps/bin:/home/sascha/.nix-profile/bin
    :/home/sascha/.poetry/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/sascha/.zgen/unixorn/jpb.zshplugin-master/bin:/home/sascha/.zgen/unixorn/warhol.plugin.zsh-master/b
    in:/home/sascha/.zgen/unixorn/git-extra-commands-master/bin:/home/sascha/.zgen/unixorn/bitbucket-git-helpers.plugin.zsh-master/bin:/home/sascha/.zgen/skx/sysadmin-util-master:/home/sascha/.zgen/StackExchange/blackbox-master/bin:/h
    ome/sascha/.yarn/bin:/usr/lib/oracle/19.6/client64/bin"
    TERM: "xterm-256color"
    USER: "sascha"
  }

User Config
  unset

Rome Version
  10.0.3-beta

Node Version
  12.18.3

Platform
  linux x64 5.4.0-7634-generic

Terminal Features
  Object {
    background: "unknown"
    colorDepth: 24
    columns: 238
    cursor: true
    hyperlinks: true
    isTTY: true
    unicode: true
  }

Client Flags
  Object {
    clientName: "cli"
    cwd: "/home/sascha/workspace/rome-bug"
    realCwd: "/home/sascha/workspace/rome-bug"
    silent: false
  }

Server Status
  Object {
    projects: Array [
      Object {id: 0}
      Object {id: 1}
    ]
    server: Object {
      heapTotal: 112_394_240
      pid: 89_969
      uptime: 1.035969301
    }
    workers: Array [
      Object {
        astCacheSize: 0
        heapTotal: 112_394_240
        ownedBytes: 0
        ownedFileCount: 0
        pid: 89_969
        uptime: 1.035625484
      }
    ]
  }
```

## Steps To Reproduce

See https://github.com/Sascha8a/rome-decorator-bug

1. Create some class with a TypeScript class Decorator
2. `rome check`
3. Class is not recognized as being exported

## Expected Results

Class is exported.

**Comment by dibenso** [comment]
https://github.com/romefrontend/rome/blob/765e91ccc1b27eff59a6205e64e23b0df66db0f3/internal/js-parser/parser/expression.ts#L1516 at a quick glance shows there is no case for handling decorators? It seems to fall into the default clause of the switch with diagnostic: `descriptions.JS_PARSER.UNKNOWN_EXPRESSION_ATOM_START`

**Comment by sebmck** [comment]
Yeah we have no support at all for decorators. They were one of the syntax extensions I removed years ago when I forked `js-parser` from `babel-parser`.

If the syntax of decorators has been settled then I'm open to adding support. I just recall a high amount of chur as the syntax was still evolving. We would also only support the latest decorator syntax revision, rather than also supporting the older legacy one like Babel.

**Comment by ematipico** [comment]
The thing is, TypeScript have decorators too (legacy version) and there are various libs that rely on them. Example:

- Mobx (frontend)
- Nestjs
- TypeORM

The big problem with decorators is that the syntax is still under revision and the proposal is still idle at stage 2 for quite long time.

**Comment by Sascha8a** [comment]
I'm not expecting there to be full support for TypeScript Decorators, however would love it for them to be just "ignored". In this case, the class should be exported normally.

**Comment by jamiebuilds** [comment]
Hi! Thank you for opening this issue

We recently made the decision to [migrate the Rome codebase to Rust](https://rome.tools/blog/2021/09/21/rome-will-be-rewritten-in-rust) and making some significant changes to its architecture. We've also decided not to release the previous version written in TypeScript.

We're going through and closing pull requests and issues that were specific to the TypeScript codebase, including this one. Thank you for taking the time to open it.

Thanks again!
